api/apps/v1alpha1/nemo_datastore_types.go

```diff
@@ -429,15 +663,88 @@ func (n *NemoDatastore) GetDefaultStartupProbe() *corev1.Probe {
 	return &probe
 }
 
-// GetVolumes returns volumes for the NemoDatastore container
-func (n *NemoDatastore) GetVolumes() []corev1.Volume {
-	volumes := []corev1.Volume{}
-	return volumes
-}
-
 // GetVolumeMounts returns volumes for the NemoDatastore container
 func (n *NemoDatastore) GetVolumeMounts() []corev1.VolumeMount {
-	return []corev1.VolumeMount{}
+	return []corev1.VolumeMount{
+		{
+			MountPath: "/tmp",
+			Name:      "temp",
+		},
+		{
+			MountPath: "/data",
+			Name:      "data",
+		},
+	}
+}
+
+func (n *NemoDatastore) GetVolumeMountsInitContainer() []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		{
+			MountPath: "/usr/sbin",
+			Name:      "config",
+		},
+		{
+			MountPath: "/tmp",
+			Name:      "temp",
+		},
+		{
+			MountPath: "/data",
+			Name:      "data",
+		},
+		{
+			MountPath: "/env-to-ini-mounts/inlines/",
+			Name:      "inline-config-sources",
+		},
+		{
+			MountPath: "/usr/sbin/init",
+			Name:      "init",
+		},
+	}
+}
+
+func (n *NemoDatastore) GetInitContainers() []corev1.Container {
+	user := int64(1000)
```

Comment:
here, we can use the same RunAsUser config from the main container if provided. If not set then default to 1000. The reason in in some distributions, the namespace is limited to certain range of UIDs and this might fail then.